lib/src/rules/missing_conditional_import.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Missing conditional import.';
+
+const _details = r'''
+
+**DON'T** reference files that do not exist in conditional imports.
+
+Code may fail at runtime if the condition evaluates such that the missing file
+needs to be imported.
+
+**BAD:**
+```dart
+import 'file_that_does_not_exist.dart';
```

Comment:
Consider updating to add a condition.

Maybe something like

```dart
import 'file_that_does_exist.dart' 
  if (condition) 'file_that_does_not_exist.dart';
```

(Or similar.)

Reply:
Oof, good call - those examples were definitely bad. Updated!